lms/djangoapps/instructor/views/api.py

```diff
@@ -3322,24 +3322,205 @@ def _instructor_dash_url(course_key, section=None):
+
+    Returns:
+        dict: A dictionary with a success message and the task ID.
+    """
+    course_key = CourseKey.from_string(course_id)
+    task = task_api.generate_certificates_for_students(request, course_key)
+
+    return {
+        "message": _(
+            "Certificate generation task for all students of this course has been started. "
+            "You can view the status of the generation task in the \"Pending Tasks\" section."
+        ),
+        "task_id": task.task_id
+    }
+
+
+def start_certificate_regeneration(request, course_id, certificates_statuses):
+    """
+    Initiates regeneration of certificates for students based on given certificate statuses.
+
+    This function triggers a background task that regenerates certificates for students
+    whose certificates match the provided list of statuses.
+
+    Args:
+        request (HttpRequest): The HTTP request object.
+        course_key (CourseKey): The identifier of the course for which certificates are being regenerated.
+        certificates_statuses (list[str]): A list of certificate statuses to filter the affected certificates.
+
+    Returns:
+        dict: A dictionary with a success message and success status.
+    """
+    course_key = CourseKey.from_string(course_id)
+    task_api.regenerate_certificates(request, course_key, certificates_statuses)
+
+    return {
+        'message': _(
+            'Certificate regeneration task has been started. '
+            'You can view the status of the generation task in the "Pending Tasks" section.'
+        ),
+        'success': True
+    }
+
+
+@method_decorator(cache_control(no_cache=True, no_store=True, must_revalidate=True), name='dispatch')
+@method_decorator(transaction.non_atomic_requests, name='dispatch')
+class CertificateTask(DeveloperErrorViewMixin, APIView):
+    """
+    API endpoint for handling certificate-related administrative tasks for a given course.
+
+    Supported actions:
+    - "toggle": Enable or disable self-generated certificates.
+    - "generate": Initiate certificate generation for all enrolled students.
+    - "regenerate": Regenerate certificates based on selected certificate statuses.
+
+    URL pattern:
+        POST /courses/{course_id}/instructor/api/certificates/{action}/
+
+    The `action` path parameter determines the task to perform.
+    The request must be authenticated and the user must have the appropriate permission for the action.
+    """
+    permission_classes = [IsAuthenticated, HasCertificateActionPermission]
+
+    @method_decorator(ensure_csrf_cookie)
+    def post(self, request, course_id, action=None):
+        """
+        Handles POST requests for certificate actions.
+
+        Args:
+            request (HttpRequest): The HTTP request object.
+            course_id (str): The ID of the course to operate on.
+            action (str): The certificate task to perform ("toggle", "generate", or "regenerate").
```

Comment:
Based on the action, you might be required to provide more params, eg. if you choose the `regenerate` action, you should provide a `statuses` list of status that you want to regenerate.  This should be documented in this docstring so that it will show up in the API docs.